pyccel/parser/syntax/headers.py

```diff
@@ -72,7 +72,8 @@ def expr(self):
         if any(isinstance(d_type, FuncType) for d_type in self.dtypes):
             msg = 'Functions in a template are not supported yet'
             errors.report(msg,
-                        severity='fatal')
+                        severity='error')
+            return EmptyNode()
```

Comment:
Why do you need an `EmptyNode` here, instead of writing `return errors.report(msg, severity='error')`?

Reply:
Added explanation: d63e27fb

Reply:
Hmm, actually I've just had a look at this and `None` is the one other value that is acceptable. Would you prefer I use `return None`?

Reply:
Yes. We do not use `EmptyNode` extensively: https://github.com/pyccel/pyccel/search?q=EmptyNode

Reply:
fixed [54e8ab2](https://github.com/pyccel/pyccel/pull/1284/commits/54e8ab2188091f2da7c023fc59d2286ce42ef32c)

Reply:
Sorry but I was probably too quick with my comment. According to the [docstring of `EmptyNode`](https://github.com/pyccel/pyccel/blob/261c1d9027efc3e7a2c33b6a1e7c081c48f0a602/pyccel/ast/core.py#L3651-L3674), objects of this class should be used as much as possible instead of `None` :sweat_smile:

Reply:
Oops, I removed the commit